src/components/Markdown.tsx

```diff
@@ -17,20 +22,56 @@ const unifiedPlugins: { remark?: PluggableList; rehype: PluggableList } = {
 interface MarkdownProps {
   /** Stringified markdown. */
   content: string;
+  /** The response ID correlates to the current message. */
+  responseId?: string;
+  /** Classnames for the container. */
+  className?: string;
 }
 
 /**
  * Renders Github-Flavored Markdown from the Knowledge Graph. This Markdown can include
  * arbitrary HTML. Any HTML will be sanitized according to Rehype's default Schema.
  *
- * @public
+ * @remarks
+ * A link click will send a CHAT_LINK_CLICK analytics event
+ *
+ * @internal
  */
-export function Markdown({ content }: MarkdownProps) {
+export function Markdown({ content, responseId, className }: MarkdownProps) {
+  const action = useChatActions();
+  const components: ReactMarkdownOptions["components"] = useMemo(() => {
+    const createClickHandlerFn = (href?: string) => () => {
+      action.report({
+        action: "CHAT_LINK_CLICK",
+        destinationUrl: href,
+        chat: {
+          responseId,
+        },
+      });
+    };
+    return {
+      a: ({ node: _, children, ...props }) => {
+        return (
+          <a
+            {...props}
+            onClick={createClickHandlerFn(props.href)}
+            target="_blank"
```

Comment:
you'll want `rel="noopener noreferrer"` here almost certainly

Reply:
done!